src/UhdmAst.cpp

```diff
@@ -2988,6 +2987,23 @@ AstNode* visit_object(vpiHandle obj_h, UhdmShared& shared) {
         });
         return new AstForeach(make_fileline(obj_h), arrayp, bodyp);
     }
+    case vpiForever: {
+        AstNode* bodyp = nullptr;
+        visit_one_to_one({vpiStmt}, obj_h, shared, [&](AstNode* itemp) {
+            bodyp = itemp;
+        });
+        return new AstWhile(make_fileline(obj_h),
+                            new AstConst(make_fileline(obj_h), AstConst::BitTrue()),
+                            bodyp);
+    }
+    case vpiDisable: {
+        std::string scopeName = "";
+        if (auto parent_h = vpi_handle(vpiParent, obj_h)) {
+            scopeName = get_object_name(parent_h, {vpiName, vpiFullName});
```

Comment:
This will work if a block disables itself, but but `disable` statement can also disable a task or another named block. We don't need to support that right now, but this code would lead to unexpected behavior.

Reply:
I will handle `disable` node when the Surelog supports it. Currently the error is thrown.

Reply:
Throwing error is fine. Please also specify in a comment that we're waiting for Surelog/UHDM to fix that.

Reply:
Done